lib/src/sync-compiler.ts

```diff
@@ -45,20 +78,115 @@ export class SyncEmbeddedCompiler {
         return true;
 
       case 'exit':
-        this.exited = true;
+        this.disposed = true;
         return false;
     }
   }
 
   /** Blocks until the underlying process exits. */
-  yieldUntilExit(): void {
-    while (!this.exited) {
+  private yieldUntilExit(): void {
+    while (!this.disposed) {
       this.yield();
     }
   }
 
-  /** Kills the child process, cleaning up all associated Observables. */
-  close() {
+  /**
+   * Sends a compile request to the child process and returns the CompileResult.
+   * Throws if there were any protocol or compilation errors.
+   */
+  private compileRequestSync(
+    request: proto.InboundMessage_CompileRequest,
+    importers: ImporterRegistry<'sync'>,
+    options?: OptionsWithLegacy<'sync'>
+  ): CompileResult {
+    const functions = new FunctionRegistry(options?.functions);
+
+    const dispatcher = createDispatcher<'sync'>(
+      this.compilationId++,
+      this.messageTransformer,
+      {
+        handleImportRequest: request => importers.import(request),
+        handleFileImportRequest: request => importers.fileImport(request),
+        handleCanonicalizeRequest: request => importers.canonicalize(request),
+        handleFunctionCallRequest: request => functions.call(request),
+      }
+    );
+    this.dispatchers.add(dispatcher);
+
+    dispatcher.logEvents$.subscribe(event => handleLogEvent(options, event));
+
+    let error: unknown;
+    let response: proto.OutboundMessage_CompileResponse | undefined;
+    dispatcher.sendCompileRequest(request, (error_, response_) => {
+      this.dispatchers.delete(dispatcher);
+      if (this.dispatchers.size === 0) this.compilationId = 1;
```

Comment:
I like matching the logic between async and sync here even if it's not strictly necessary in sync mode.